x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/helpers.test.ts

```diff
@@ -45,16 +45,64 @@ describe('helpers', () => {
       expect(getNumOfCoveredSubtechniques(payload)).toEqual(0);
     });
 
-    it('returns number of subtechniques when present', () => {
+    it('returns total number of unique enabled and disabled subtechniques when no filter is passed', () => {
       const payload = {
         ...getMockCoverageOverviewMitreTechnique(),
         subtechniques: [
           getMockCoverageOverviewMitreSubTechnique(),
-          getMockCoverageOverviewMitreSubTechnique(),
+          { ...getMockCoverageOverviewMitreSubTechnique(), id: 'test-id' },
         ],
       };
       expect(getNumOfCoveredSubtechniques(payload)).toEqual(2);
     });
+
+    it('returns total number of unique enabled and disabled subtechniques when both filters are passed', () => {
+      const payload = {
+        ...getMockCoverageOverviewMitreTechnique(),
+        subtechniques: [
+          getMockCoverageOverviewMitreSubTechnique(),
+          { ...getMockCoverageOverviewMitreSubTechnique(), id: 'test-id' },
+        ],
+      };
+      expect(
+        getNumOfCoveredSubtechniques(payload, [
+          CoverageOverviewRuleActivity.Enabled,
+          CoverageOverviewRuleActivity.Disabled,
+        ])
+      ).toEqual(2);
+    });
+
+    it('returns total number of enabled subtechniques when enabled filter is passed', () => {
+      const payload = {
```

Comment:
Nit: Could you please add explicit type annotations to all the payload variables?

```ts
const payload: CoverageOverviewMitreTechnique = { ...
```